crates/libcgroups/src/common.rs

```diff
@@ -173,40 +176,78 @@ pub fn create_cgroup_manager<P: Into<PathBuf>>(
     container_name: &str,
 ) -> Result<Box<dyn CgroupManager>> {
     let cgroup_setup = get_cgroup_setup()?;
+    let cgroup_path = cgroup_path.into();
 
     match cgroup_setup {
-        CgroupSetup::Legacy | CgroupSetup::Hybrid => {
-            log::info!("cgroup manager V1 will be used");
-            Ok(Box::new(v1::manager::Manager::new(cgroup_path.into())?))
-        }
+        CgroupSetup::Legacy | CgroupSetup::Hybrid => create_v1_cgroup_manager(cgroup_path),
         CgroupSetup::Unified => {
             if systemd_cgroup {
-                if !systemd::booted() {
-                    bail!("systemd cgroup flag passed, but systemd support for managing cgroups is not available");
-                }
-
-                let use_system = nix::unistd::geteuid().is_root();
-
-                log::info!(
-                    "systemd cgroup manager with system bus {} will be used",
-                    use_system
-                );
-                return Ok(Box::new(systemd::manager::Manager::new(
-                    DEFAULT_CGROUP_ROOT.into(),
-                    cgroup_path.into(),
-                    container_name.into(),
-                    use_system,
-                )?));
+                return create_systemd_cgroup_manager(cgroup_path, container_name);
             }
-            log::info!("cgroup manager V2 will be used");
-            Ok(Box::new(v2::manager::Manager::new(
-                DEFAULT_CGROUP_ROOT.into(),
-                cgroup_path.into(),
-            )?))
+
+            create_v2_cgroup_manager(cgroup_path)
         }
     }
 }
 
+#[cfg(feature = "v1")]
+fn create_v1_cgroup_manager(cgroup_path: PathBuf) -> Result<Box<dyn CgroupManager>> {
+    log::info!("cgroup manager V1 will be used");
+    Ok(Box::new(v1::manager::Manager::new(cgroup_path)?))
+}
+
+#[cfg(not(feature = "v1"))]
+fn create_v1_cgroup_manager(_cgroup_path: PathBuf) -> Result<Box<dyn CgroupManager>> {
+    bail!("cgroup v1 feature is not activated");
```

Comment:
Maybe for these features we should add some context around the fact that the build excluded this feature? I know we didn't do this in the past, but maybe it's worth doing moving forward.

Reply:
I really appreciate this being broken out and put behind feature flags. I think this is much cleaner.